app/src/main/java/com/fluffybros/tujefe/ui/add/ManualAddFragment.kt

```diff
@@ -1,6 +1,29 @@
 package com.fluffybros.tujefe.ui.add
 
+import android.app.Activity
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.InputMethodManager
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
+import androidx.navigation.fragment.findNavController
+import com.fluffybros.tujefe.MainViewModel
 import com.fluffybros.tujefe.R
+import com.fluffybros.tujefe.databinding.FragmentManualAddBinding
 
-class ManualAddFragment : Fragment(R.layout.fragment_manual_add)
+class ManualAddFragment : Fragment(R.layout.fragment_manual_add) {
+    private val mainViewModel: MainViewModel by activityViewModels()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val binding = FragmentManualAddBinding.bind(view)
+
+        binding.manualAddButton.setOnClickListener {
+            mainViewModel.addRecyclerItem(binding.nameField.editText?.text.toString(), binding.setupKeyField.editText?.text.toString())
+            val imm: InputMethodManager =
+                requireContext().getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
```

Comment:
casting is generally not great, since it's prone to runtime errors. you can use a kotlin extension function to do this instead - 

https://android.jlelse.eu/kotlin-extensions-for-android-part-1-32aeb276b6b2